server/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -148,6 +147,18 @@ public boolean hasIndexBlock(String index, ClusterBlock block) {
         return indicesBlocks.containsKey(index) && indicesBlocks.get(index).contains(block);
     }
 
+    public boolean hasIndexBlock(String index, int blockId) {
```

Comment:
perhaps rename this to `hasIndexBlockWithId`?

Reply:
Should I also rename `hasGlobalBlock(int blockId)` to `hasGlobalBlockWithId(int blockId)` in a follow up PR against master ?

Reply:
yes, either here or in a follow-up.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/36941